### PR TITLE
[CCI] Add bluebird replaces for packages/osd-opensearch-archiver

### PR DESCRIPTION
**Pull request by Nicksqain**
### Issues Resolved

- A partial solution #3662

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: packages/osd-opensearch-archiver/src/actions/rebuild_all.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ import {
 } from '../lib';
 
 async function isDirectory(path: string): Promise<boolean> {
-  const stats: Stats = await fromNode((cb) => stat(path, cb));
+  const stats: Stats = await promisify(stat)(path);
```
Comment (AMoo-Miki) [inline packages/osd-opensearch-archiver/src/actions/rebuild_all.ts:47]: Should use `fs/promises`:

```js
import { stat } from 'fs/promises';
```
```suggestion
  const stats: Stats = await stat(path);
```

File: packages/osd-opensearch-archiver/src/actions/rebuild_all.ts [hunk 2]
```diff
@@ -82,7 +82,7 @@ export async function rebuildAllAction({
       createWriteStream(tempFile),
     ] as [Readable, ...Writable[]]);
 
-    await fromNode((cb) => rename(tempFile, childPath, cb));
+    await promisify(rename)(tempFile, childPath);
```
Comment (AMoo-Miki) [inline packages/osd-opensearch-archiver/src/actions/rebuild_all.ts:85]: Should use `fs/promises` as well.

File: packages/osd-opensearch-archiver/src/lib/directory.ts [hunk 3]
```diff
@@ -29,9 +29,9 @@
  */
 
 import { readdir } from 'fs';
-import { fromNode } from 'bluebird';
+import { promisify } from 'util';
 
 export async function readDirectory(path: string) {
-  const allNames = await fromNode<string[]>((cb) => readdir(path, cb));
+  const allNames: string[] = await promisify(readdir)(path);
```
Comment (AMoo-Miki) [inline packages/osd-opensearch-archiver/src/lib/directory.ts:35]: Should use `fs/promises` as well.

File: packages/osd-opensearch-archiver/src/lib/docs/generate_doc_records_stream.test.ts [hunk 4]
```diff
@@ -30,14 +30,17 @@
...
 import { createListStream, createPromiseFromStreams, createConcatStream } from '../streams';
 
 import { createGenerateDocRecordsStream } from './generate_doc_records_stream';
 import { Progress } from '../progress';
 import { createStubStats, createStubClient } from './test_stubs';
 
+function delay(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
```
Comment (AMoo-Miki) [inline packages/osd-opensearch-archiver/src/lib/docs/generate_doc_records_stream.test.ts:43]: We have been defining this `delay` function way too often. We should move this out into an importable.

**Review by AMoo-Miki — CHANGES_REQUESTED** [review]

File: packages/osd-opensearch-archiver/src/lib/docs/generate_doc_records_stream.test.ts (hunk 4 above)
Comment (Nicksqain) [reply packages/osd-opensearch-archiver/src/lib/docs/generate_doc_records_stream.test.ts:43]: @AMoo-Miki 
Thank you for ur review! I was wondering about that, so maybe you can tell me where I can place the `delay` function?
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3662#issuecomment-1503893712

File: packages/osd-opensearch-archiver/src/actions/rebuild_all.ts (hunk 1 above)
Comment (Nicksqain) [reply packages/osd-opensearch-archiver/src/actions/rebuild_all.ts:47]: Sorry, I was sticking to description #3662
I'll look into it

File: packages/osd-opensearch-archiver/src/lib/docs/generate_doc_records_stream.test.ts (hunk 4 above)
Comment (AMoo-Miki) [reply packages/osd-opensearch-archiver/src/lib/docs/generate_doc_records_stream.test.ts:43]: `packages/osd-utils` would be a perfect spot; just throw it into a new file. This way when we get rid of Node 14, we can just use `timers/promises` and delete this file.

File: packages/osd-opensearch-archiver/src/actions/rebuild_all.ts (hunk 1 above)
Comment (AMoo-Miki) [reply packages/osd-opensearch-archiver/src/actions/rebuild_all.ts:47]: Don't apologize. I should thank you for doing all of this!

**Comment by abbyhu2000** [comment]
convert to draft since requested changes have not been addressed for more than two weeks

**Comment by ananzh** [comment]
@Nicksqain Due to an extended period without updates, we're going to close this issue for now. We appreciate your contribution and understand that priorities and availability can change. Please feel free to reopen this issue when you have the opportunity to revisit it. Thank you for your understanding.

**Pull request closed without merging**